src/player.js

```diff
@@ -1517,7 +1517,10 @@ export default class Player extends FakeEventTarget {
           const videoElement = this.getVideoElement();
           return videoElement ? Array.from(videoElement.textTracks).findIndex(track => track ? track.language === textTrack.language : false) : -1;
         };
-        this._getTracksByType(TrackType.TEXT).forEach(track => track.index = getNativeLanguageTrackIndex(track));
+        this.ready().then(() => {
```

Comment:
If we already need to wait for ready event to map the index than it probably means that we don’t need to be handled here at all and move this to the ready promise chain just before we emit the regular TEXT_TRACK_CHANGED

Reply:
@OrenMe might there be a case when the ready promise will resolved before we add native tracks?

Reply:
External native tracks are controlled by you and not by 3rd party lib that parses and downloads a manifest so this is something you have full control over. AFAICT no need for guessing work here

Reply:
@odedhutzler what is the status of this?